### PR TITLE
Add example of grpc server and client

### PR DESCRIPTION
**Pull request by andriikushch**

### CODE REVIEW

File: example/grpc-example/main.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+)
+
+const (
+	defaultPort    = ":43210"
+	defaultAddress = "localhost"
+	testMessage    = "Hi!"
+)
+
+func main() {
+	// `port` will be used by an example
```
Comment (andrewslotin) [inline example/grpc-example/main.go:30]: Let's omit the comments that are not related to instrumentation with Instana

File: example/grpc-example/main.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// Service is used to implement pb.EchoService
+type Service struct {
+	pb.UnimplementedEchoServiceServer
+}
+
+// Echo implements pb.EchoService
+func (s *Service) Echo(ctx context.Context, in *pb.EchoRequest) (*pb.EchoReply, error) {
+	log.Printf("Request << %s", in.GetMessage())
+
+	// Extract the parent span and use its tracer to initialize any child spans to trace the calls
```
Comment (andrewslotin) [inline example/grpc-example/main.go:129]: Let's emphasize that this is purely optional and not required for minimal instrumentation

**Review by andrewslotin — CHANGES_REQUESTED** [review]
Overall looks good! I think we could simplify this example even more and make sure that we only keep comments that are specific to instrumentation. This is code is to demonstrate the use of Instana after all, and not how to implement a gRPC server 😃

File: example/grpc-example/main.go (hunk 1 above)
Comment (andriikushch) [reply example/grpc-example/main.go:30]: Makes sense, I will change it.

File: example/grpc-example/main.go (hunk 2 above)
Comment (andriikushch) [reply example/grpc-example/main.go:105]: This is done.

File: example/grpc-example/main.go (hunk 1 above)
Comment (andriikushch) [reply example/grpc-example/main.go]: This is done.

**Review by andrewslotin — APPROVED** [review]
👍
